core/shared/src/main/scala/fs2/hashing/Hashing.scala

```diff
@@ -26,50 +26,90 @@ import cats.effect.{IO, LiftIO, Resource, Sync, SyncIO}
 
 /** Capability trait that provides hashing.
   *
-  * The [[create]] method returns an action that instantiates a fresh `Hash` object.
-  * `Hash` is a mutable object that supports incremental computation of hashes. A `Hash`
-  * instance should be created for each hash you want to compute (`Hash` objects may be
-  * reused to compute multiple hashes but care must be taken to ensure no concurrent usage).
+  * The [[hash]] method returns a fresh `Hash` object as a resource. `Hash` is a
+  * mutable object that supports incremental computation of hashes.
+  *
+  * A `Hash` instance should be created for each hash you want to compute, though `Hash`
+  * objects may be reused to compute consecutive hashes. When doing so, care must be taken
+  * to ensure no concurrent usage.
+  *
+  * The `hashWith` operation converts a `Resource[F, Hash[F]]` to a `Pipe[F, Byte, Digest]`.
+  * The resulting pipe outputs a single `Digest` once the source byte stream terminates.
+  *
+  * Alternatively, a `Resource[F, Hash[F]]` can be used directly (via `.use` or via
+  * `Stream.resource`). The `Hash[F]` trait provides lower level operations for computing
+  * hashes, both at an individual chunk level (via `update` and `digest`) and at stream level
+  * (e.g., via `observe` and `drain`).
+  *
+  * Finally, the `Hashing` companion object offers utilities for computing pure hashes:
+  * `hashPureStream` and `hashChunk`.
   */
 sealed trait Hashing[F[_]] {
 
   /** Creates a new hash using the specified hashing algorithm. */
-  def create(algorithm: HashAlgorithm): Resource[F, Hash[F]]
+  def hash(algorithm: HashAlgorithm): Resource[F, Hash[F]]
 
   /** Creates a new MD-5 hash. */
-  def md5: Resource[F, Hash[F]] = create(HashAlgorithm.MD5)
+  def md5: Resource[F, Hash[F]] = hash(HashAlgorithm.MD5)
```

Comment:
Undecided if we should have all these aliases. Options:
- No aliases -- use `Hashing[F].hash(HashAlgorithm.SHA256)` instead
- Aliases for most popular hashes only -- SHA1, SHA256, SHA3_256?
- Aliases for all built-in hashes (that's current state of this PR)

We don't have aliases for HMACs (though we could) -- e.g., `def hmacSha256(key: Chunk[Byte]) = hmac(hash(HashAlgorithm.SHA256), key)`

Thoughts @armanbilge?

Reply:
> * Aliases for all built-in hashes

It's not really possible to have this in general. By "built-in" we mean hashes that we have a `case` for. Because what's actually built-in on the runtime system can be totally variable depending on the build of the JDK/Node.js/OpenSSL. In truth, any of these methods could raise an error. Maybe no aliases would be more honest to that, although I can see how that's annoying from a usability perspective.

Reply:
Added some docs to make the possibility of failure clear and removed all aliases.